pkg/controllermanager/controller/project/project_control_delete.go

```diff
@@ -31,36 +32,46 @@ import (
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/controllerutils"
 	"github.com/gardener/gardener/pkg/operation/common"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
 )
 
-func (r *projectReconciler) delete(ctx context.Context, project *gardencorev1beta1.Project, gardenClient kubernetes.Interface) (reconcile.Result, error) {
+func (r *projectReconciler) delete(ctx context.Context, project *gardencorev1beta1.Project, gardenClient client.Client, gardenAPIReader client.Reader) (reconcile.Result, error) {
 	if namespace := project.Spec.Namespace; namespace != nil {
-		isEmpty, err := isNamespaceEmpty(ctx, gardenClient.APIReader(), *namespace)
+		isEmpty, err := isNamespaceEmpty(ctx, gardenAPIReader, *namespace)
 		if err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to check if namespace is empty: %w", err)
 		}
 
 		if !isEmpty {
 			r.reportEvent(project, true, gardencorev1beta1.ProjectEventNamespaceNotEmpty, "Cannot release namespace %q because it still contains Shoots.", *namespace)
-			_, _ = updateProjectStatus(ctx, gardenClient.GardenCore(), project.ObjectMeta, setProjectPhase(gardencorev1beta1.ProjectTerminating))
+			_ = kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, gardenClient, project, func() error {
+				project.Status.Phase = gardencorev1beta1.ProjectTerminating
+				return nil
+			})
```

Comment:
We not directly switch to a patch? :)

Reply:
OK, I've switched to `TryPatchStatus`, I hope this is what you had in mind.